### PR TITLE
Support ed25519 2020

### PR DESCRIPTION
**Pull request by JSAssassin**
~~Note: Some tests has been skipped since the documentLoader currently doesn't have did:v1 resolver added to it yet. All those tests pass though when I change the didMethod from v1 to key. 
So whenever veres one resolver gets added to the documentLoader, these tests can be unskipped.~~

Update: All tests are passing now and documentLoader has been updated.

### CODE REVIEW

File: ProfileManager.js [hunk 1]
```diff
@@ -650,6 +650,7 @@ export default class ProfileManager {
     assert.parentCapabilitiesValidator({
       parentCapabilities, edvId, hmac, keyAgreementKey
     });
+    console.log('gets here:::::::', invocationSigner);
```
Comment (mattcollier) [inline ProfileManager.js:653]:
```suggestion
```

File: ProfileManager.js [hunk 2]
```diff
@@ -695,15 +696,15 @@ export default class ProfileManager {
...
+        publicAlias: keyAgreementKey.id
       };
       delegateEdvKakRequest.parentCapability = keyAgreementKey.id;
     } else {
       const {keyAgreementKey: kak} = parentCapabilities;
       delegateEdvKakRequest.invocationTarget = {...kak.invocationTarget};
       delegateEdvKakRequest.parentCapability = kak;
     }
-
+    console.log(invocationSigner, '<><><><>invocationSigner');
```
Comment (mattcollier) [inline ProfileManager.js:707]:
```suggestion
```

**Review by mattcollier — CHANGES_REQUESTED** [review]
Needs a changelog entry.  Looks like we're effectively removing veres-one DID support temporarily, so that will need to be listed as a breaking change.

**Comment by JSAssassin** [comment]
Note: There should be an additional commit that removes the logs and unused dep. It's not showing up on the UI for some reason, I think github is slow or something. I can see the commit when I click on the branch though.

**Review by dmitrizagidulin — APPROVED** [review]
:+1: 
Nicely done

File: utils.js [hunk 3]
```diff
@@ -240,17 +265,25 @@ export async function id() {
...
+  if(signer.type === 'Ed25519VerificationKey2018') {
+    suite = new Ed25519Signature2018({
       signer,
       verificationMethod: signer.id
-    }),
+    });
+  } else if(signer.type === 'Ed25519VerificationKey2020') {
+    suite = new Ed25519Signature2020({
+      signer,
+      verificationMethod: signer.id
```
Comment (dmitrizagidulin) [inline utils.js:277]:
```suggestion
```
(the suites no longer have a `verificationMethod` constructor param).

File: utils.js [hunk 4]
```diff
@@ -240,17 +265,25 @@ export async function id() {
...
     Array.isArray(capabilityChain) ? capabilityChain : [zcap.parentCapability];
-  // attach capability delegation proof
-  return sign(zcap, {
-    // TODO: map `signer.type` to signature suite
-    suite: new Ed25519Signature2018({
+  let suite;
+  if(signer.type === 'Ed25519VerificationKey2018') {
+    suite = new Ed25519Signature2018({
       signer,
       verificationMethod: signer.id
```
Comment (dmitrizagidulin) [inline utils.js:272]:
```suggestion
```

File: utils.js (hunk 3 above)
Comment (JSAssassin) [reply utils.js]: Addressed xa1f58af

**Comment by JSAssassin** [comment]
Waiting on:

- [ ] https://github.com/digitalbazaar/bedrock-kms/pull/41
- [ ] https://github.com/digitalbazaar/bedrock-kms-http/pull/51
- [ ] https://github.com/digitalbazaar/bedrock-profile/pull/68
- [ ] https://github.com/digitalbazaar/bedrock-edv-storage/pull/77
- [ ] https://github.com/digitalbazaar/bedrock-did-io/pull/3
- [ ] https://github.com/digitalbazaar/bedrock-profile-http/pull/34
- [ ] https://github.com/digitalbazaar/bedrock-web-profile/pull/16

File: ProfileManager.js [hunk 5]
```diff
@@ -1148,5 +1147,6 @@ async function _createCapabilityAgent({handle}) {
   const secret = new Uint8Array(32);
   crypto.getRandomValues(secret);
 
-  return CapabilityAgent.fromSecret({secret, handle});
+  return CapabilityAgent.fromSecret({
+    secret, handle, keyType: 'Ed25519VerificationKey2020'});
```
Comment (mattcollier) [inline ProfileManager.js:1151]:
```suggestion
  return CapabilityAgent.fromSecret({
    secret, handle, keyType: 'Ed25519VerificationKey2020'
  });
```

**Review by mattcollier — APPROVED** [review]
TY, more more style nit.

File: utils.js [hunk 6]
```diff
@@ -240,17 +266,23 @@ export async function id() {
...
+  let suite;
+  if(signer.type === 'Ed25519VerificationKey2018') {
+    suite = new Ed25519Signature2018({
+      signer,
+    });
+  } else if(signer.type === 'Ed25519VerificationKey2020') {
+    suite = new Ed25519Signature2020({
+      signer,
+    });
+  }
```
Comment (mattcollier) [inline utils.js:278]: I just noticed, that we should throw an error if we hit the bottom without finding a matching `signer.type`.

**Comment by mattcollier** [comment]
tracking: https://github.com/digitalbazaar/bedrock-karma/issues/14

**Pull request merged**